### DISCUSSION TITLE
7530 AX push-firmware geht nicht

### DISCUSSION THREAD

**Discussion by feniarus** [Freetz-NG/freetz-ng#628] [push firmware]
Hab das Image nach Anleitung gebaut und versucht mehrfach zu flashen über make push-fireware ohne erfolg.

Log:
![Bildschirmfoto vom 2022-10-05 04-36-39](https://user-images.githubusercontent.com/101282511/193969218-e27699ff-954e-4413-b373-3a147a9b3760.png)

Konfig:
[config.txt](https://github.com/Freetz-NG/freetz-ng/files/9712159/config.txt)

PS: Bitte dieses mal nicht gleich wieder schließen ohne wirklich mir geholfen zu haben wie zuvor ....

**Comment by fda77** [comment]
Hast du nicht bemerkt dass auf deinen 2 Screenshots unterschiedliche Meldungen sind?!?! Hab keine Ahnung von welcher Anleitung du redest

**Reply by feniarus** [reply]
Na von der Kurzanleitung Auf der Startseite vom GitHub Projekt.
Hab es mit zwei verschiedenen Boxen versucht daher die beiden einzelnen Fälle bzw. Screenshot.

**Reply by fda77** [reply]
Logisch, deshlab schreibst du 1 davon in die Überschrift

**Comment by freetz-ng-mod** [comment]
Ich habe nun bestimmt schon über 40 Mal die 7530AX per push-firmware beschrieben und kann dir 100%ig sagen, dass es geht.
Als Erstes solltest du deinen Rechner eine feste IP geben.
Das mache ich immer
`IP-Adresse des Rechners auf "192.168.178.32" stellen. Gateway auf "192.168.178.1", Subnet auf 255.255.255.0`
Hier mal das log
make push-firmware
```
Freetz-NG 20290M-c405257da master 2022-10-04

 * Analyzing 'images/7530AX_07.31.all_freetz-ng-20290M-c405257da.image' ... 
 * Product: Fritz_Box_HW256 (FRITZ!Box 7530 AX)
 * Using command: ftp
 * Target host: 192.168.178.1
 * Outgoing IP: 192.168.178.32
 * Flash mode: fit-boot
 * Allowed memory size: 512 MB
 * Designated linux_fs_start: <other>

 !!! WARNING !!! WARNING !!! WARNING !!! WARNING !!! WARNING !!!
 !!!  THERE IS NO WARRANTY AT ALL !!! USE AT YOUR OWN RISK   !!!

 * Are you sure, that you want to flash this file to the device?
   images/7530AX_07.31.all_freetz-ng-20290M-c405257da.image
   Proceed? (y/[n]) y

 * You should now reboot your box (192.168.178.1). Waiting for shut down.
   Switch off, if reboot is not detected because it happens too quickly.
   Some newer bootloader versions allow to flash on power-cycle only.
   
 * MAPSTART=0x00000000
 * FULLSIZE=0x20000000	 (512 MB)
 * MAPLIMIT=0x20000000
 * FILESIZE=0x023bd61b	 (~35 MB)
 * ALIBYTES=0x000029e5	 (~10 kB)
 * FREESIZE=0x1dc40000	 (~476 MB)
 * MTDSTART=0x1dc40000

 * No reply from box, assuming switch-off or restart. Trying to re-detect box.
   Waiting ...........................ping: connect: Das Netzwerk ist nicht erreichbar
.ping: connect: Das Netzwerk ist nicht erreichbar
.ping: connect: Das Netzwerk ist nicht erreichbar
.ping: connect: Das Netzwerk ist nicht erreichbar
.ping: connect: Das Netzwerk ist nicht erreichbar
.ping: connect: Das Netzwerk ist nicht erreichbar
.. found!

 * Box is back up again, initiating transfer.

Connected to 192.168.178.1.
220 ADAM2 FTP Server ready
331 Password required for adam2
230 User adam2 successfully logged in
Remote system type is AVM.
Debugging on (debug=1).
---> TYPE I
200 Type set to BINARY
---> MEDIA SDRAM
200 Media set to MEDIA_SDRAM
---> SETENV memsize 0x1dc40000
200 SETENV command successful
---> SETENV kernel_args_tmp "avm_fwupdate mtdram1=0x1dc40000,0x20000000 mtdparts_ext=update-image.0:0x023bd61b@0x0(fit-image)"
200 SETENV command successful
local: /tmp/freetz_Fin/fitboot.flash remote: 0x1dc40000 0x20000000
ftp: setsockopt SO_DEBUG (ignored): Keine Berechtigung
---> EPSV
502 Command not implemented
disabling epsv4 for this connection
---> PASV
227 Entering Passive Mode (192,168,178,1,95,32)
---> STOR 0x1dc40000 0x20000000
150 Opening BINARY data connection
100% |******************************************************************************************************************************************************************************************************************| 36597 KiB   31.26 MiB/s    00:00 ETA226 Transfer complete
37475867 bytes sent in 00:20 (1.76 MiB/s)
---> QUIT
221 Thank you for using the FTP service on ADAM2
```

**Reply by feniarus** [reply]
Die IP fest zu vergeben hat geholfen! 
Die Fireware konnte ich hochladen aber nach dem Neustart hat sich leider nichts geändert?
Port 81 ist nicht offen und alles ist wie vor dem Hochladen.

**Reply by feniarus** [reply]
In der WebUI steht auch nicht das er die Firmware angenommen hat
![Bildschirmfoto vom 2022-10-05 23-41-40](https://user-images.githubusercontent.com/101282511/194169456-99655458-9668-4f5d-97c8-b7f3079436fd.png)

**Reply by feniarus** [reply]
Hab es hinbekommen die Box zu flashen das Image war zu groß weshalb die Box nur ein Pseudo-Update gemacht hat. 
Hatte es danach mit einem Image probiert ohne Zusatzpakete und das hat reibungslos geklappt.
Wichtig ist wie @freetz-ng-mod geschrieben hatte die IP-Adresse vom PC fest einzustellen.

Gibt es eigentlich eine Möglichkeit den Speicherverbrauch vom Image zu checken ob dieses nicht zu groß ist zum flashen?

**Reply by fda77** [reply]
```
 * FILESIZE=0x023bd61b	 (~35 MB)
```
Wieviel Platz hat dein Gerät denn?
